src/map.cpp

```diff
@@ -897,6 +899,14 @@ void Map::transformLiquids(std::map<v3s16, MapBlock*> &modified_blocks)
 			// set the liquid level and flow bit to 0
 			n0.param2 = ~(LIQUID_LEVEL_MASK | LIQUID_FLOW_DOWN_MASK);
 		}
+
+		// on_flood() the node
+		if (floodable_node != CONTENT_AIR) {
+			if (env.getScriptIface()->node_on_flood(p0, n0, new_node_content))
```

Comment:
The callback is called after `n0.param2` is modified. That means that the param2 of n0 will already be the new param2, which can cause problems with coloured nodes, for instance (like the node dropping with a different colour than it should).